### ISSUE TITLE
Junos_Config - ValueError: I/O operation on closed file

### ISSUE THREAD

**Issue by zgould** [Juniper/ansible-junos-stdlib#253]
`ansible 2.4.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/zgould/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
`

> 
> -name:
> 
>  Install JunOS
>   hosts: junos-sales-switch
>   serial: 1
>   roles:
>   - Juniper.junos
>   connection: local
>   gather_facts: no
>    netconf:
>     host: "{{ inventory_hostname }}"
>     username: admin
> 
>   tasks:
> 
>   - name: Checking NETCONF connectivity
>     wait_for: host={{ inventory_hostname }} port=830 timeout=5
> 
>   - name: Deactivate BGP Peering
>     junos_config:
>      lines:
>      - "deactivate protocols bgp group {{ BGP_GROUP }} local-address"
>      provider: "{{ netconf }}"

#############
Issue
#############

Upgrading to 2.4 Development to get some other modules working, now running into this issue with the junos_config module.

Running this in --check mode actually causes the entire play to hang until cancelled (will not timeout on it's own).

I could only get the debug information by running this live on a test machine.

````
<kc2-swi-salescloset> using connection plugin netconf
 16396 1502485464.21881: Loading Connection 'persistent' from /usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/persistent.py
 16396 1502485464.21912: Loading ShellModule 'csh' from /usr/local/lib/python2.7/dist-packages/ansible/plugins/shell/csh.py (found_in_cache=True, class_only=False)
 16396 1502485464.21921: Loading ShellModule 'fish' from /usr/local/lib/python2.7/dist-packages/ansible/plugins/shell/fish.py (found_in_cache=True, class_only=False)
 16396 1502485464.21930: Loading ShellModule 'powershell' from /usr/local/lib/python2.7/dist-packages/ansible/plugins/shell/powershell.py (found_in_cache=True, class_only=False)
 16396 1502485464.21942: Loading ShellModule 'sh' from /usr/local/lib/python2.7/dist-packages/ansible/plugins/shell/sh.py (found_in_cache=True, class_only=False)
 16396 1502485464.21955: Loading ShellModule 'sh' from /usr/local/lib/python2.7/dist-packages/ansible/plugins/shell/sh.py (found_in_cache=True, class_only=False)
 16396 1502485475.46594: done running TaskExecutor() for kc2-swi-salescloset/TASK: Deactivate BGP Peering [005056a5-5683-6455-6127-000000000047]
 16396 1502485475.46635: sending task result for task 005056a5-5683-6455-6127-000000000047
 16396 1502485475.46733: done sending task result for task 005056a5-5683-6455-6127-000000000047
 16396 1502485475.46744: WORKER PROCESS EXITING
 16365 1502485475.46940: marking kc2-swi-salescloset as failed
 16365 1502485475.46967: marking host kc2-swi-salescloset failed, current state: HOST STATE: block=2, task=2, rescue=0, always=0, run_state=ITERATING_TASKS, fail_state=FAILED_NONE, pending_setup=False, tasks child state? (None), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False
 16365 1502485475.46984: ^ failed state is now: HOST STATE: block=2, task=2, rescue=0, always=0, run_state=ITERATING_COMPLETE, fail_state=FAILED_TASKS, pending_setup=False, tasks child state? (None), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False
 16365 1502485475.47001: getting the next task for host kc2-swi-salescloset
 16365 1502485475.47016: host kc2-swi-salescloset is done iterating, returning
The full traceback is:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 125, in run
    res = self._execute()
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 526, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/action/junos_config.py", line 46, in run
    result = super(ActionModule, self).run(tmp, task_vars)
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/action/junos.py", line 79, in run
    socket_path = connection.run()
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/connection/persistent.py", line 103, in run
    out = p.stdout.readline()
ValueError: I/O operation on closed file

fatal: [kc2-swi-salescloset]: FAILED! => {
    "failed": true,
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}

```

**Comment by stacywsmith** [comment]
@zgould You are using the `junos_config` module which is part of Ansible core, not the `junos_install_config` module which is part of the `Juniper.junos` role.

Please let us know if you still see this problem with the released version of Ansible 2.4.0. If so, we'll need to open an issue with the Ansible core team to resolve.

**Comment by stacywsmith** [comment]
If this issue is still present, a new issue needs to be opened in the [Ansible repository](https://github.com/ansible/ansible/issues).
